lib/em-http/http_connection.rb

```diff
@@ -149,7 +149,11 @@ def peer
 
     def receive_data(data)
       begin
-        @p << data
+        if client.request_body_pending? && data.starts_with?('HTTP/1.1 100 Continue')
```

Comment:
This probably works.. but flaky, there is no guarantee that the received data will be a full 100 Continue response line.

Reply:
Indeed, this is very brittle, but probably sufficient for my use case (i.e. AWS S3 client).  Ideally, this would be handled by `http_parser`, but as far as I can tell, it just silently eats the '100 Continue' response instead of emitting an event of some kind.  Since this functionality already requires changes to two or three rubygems, I didn't want to get into hacking `http_parser` as well, but perhaps someone should.

Reply:
Can you add a comment to that effect in the code? We should flag this as limitation / TODO.